src/discordas.ts

```diff
@@ -214,20 +216,67 @@ async function run(): Promise<void> {
 
     roomhandler.bindThirdparty();
 
+    try {
+        await startDiscordBot(discordbot, appservice.botClient, config.bridge.adminMxid);
+        log.info("Discordbot started successfully");
+    } catch (err) {
+        log.error(err);
+        log.error("Failure during startup. Exiting");
+        process.exit(1);
+    }
+
     await appservice.begin();
     log.info(`Started listening on port ${port}`);
 
+}
+
+interface AdminRooms {
+    [mxid: string]: string;
+}
+
+async function notifyBridgeAdmin(client: MatrixClient, adminMxid: string, message: string) {
+    log.verbose(`Looking up admin room for ${adminMxid}`);
+    let adminRooms = await client.getAccountData<AdminRooms>(ADMIN_ROOMS_KEY).catch(() => null) || {};
+    if (!adminRooms[adminMxid]) {
+        log.verbose(`Creating an admin room for ${adminMxid}`);
+        adminRooms[adminMxid] = await client.createRoom();
+        await client.inviteUser(adminMxid, adminRooms[adminMxid]);
+        await client.setAccountData(ADMIN_ROOMS_KEY, adminRooms);
+    }
+    await client.sendText(adminRooms[adminMxid], message)
+}
+
+let adminNotified = false;
+
+async function startDiscordBot(
+    discordbot: DiscordBot,
+    client: MatrixClient,
+    adminMxid: string|null,
+    falloffSeconds = 5
+) {
     try {
         await discordbot.init();
         await discordbot.run();
-        log.info("Discordbot started successfully");
     } catch (err) {
-        log.error(err);
-        log.error("Failure during startup. Exiting");
-        process.exit(1);
+        if (err.code === 'TOKEN_INVALID' && adminMxid && !adminNotified) {
+            await notifyBridgeAdmin(client, adminMxid, `Your Discord bot token seems to be invalid, and the bridge cannot function. Please update it in your bridge settings and restart the bridge`);
```

Comment:
I think we should have the ability to customize this, e.g. linking to the correct page on EMS

Reply:
In in a7adca9